### ISSUE TITLE
MBHD did not shut down cleanly

### ISSUE THREAD

**Issue by jim618** [bitcoin-solutions/multibit-hd#531]
Shut down MBHD - the GUI disappears but the process (and icon in Mac OS) did not disappear

Error in log:
```
[2015-04-20 12:58:56,165] DEBUG [safe-fixed-core-events-0] o.m.h.c.m.WalletManager - Saving wallet with id : 3a331738-d6f4a2ac-0ee81449-ba205c79-c1af7472, height : 352919 
[2015-04-20 12:58:56,464] DEBUG [Wallet autosave thread] o.m.h.c.m.WalletAutoSaveListener - Auto-saving wallet to tempFile:
'/Users/jim/Library/Application Support/MultiBitHD/mbhd-3a331738-d6f4a2ac-0ee81449-ba205c79-c1af7472/wallet5594042031802469823.tmp' 
[2015-04-20 12:58:56,504] DEBUG [Wallet autosave thread] o.m.h.c.m.WalletAutoSaveListener - Wallet auto-saved to newlySavedFile:
'/Users/jim/Library/Application Support/MultiBitHD/mbhd-3a331738-d6f4a2ac-0ee81449-ba205c79-c1af7472/mbhd.wallet' 
[2015-04-20 12:58:56,839] ERROR [safe-fixed-core-events-0] o.m.h.c.e.ExceptionHandler - Cannot make encrypted copy for file '/Users/jim/Library/Application Support/MultiBitHD/mbhd-3a331738-d6f4a2ac-0ee81449-ba205c79-c1af7472/mbhd.wallet' ! org.multibit.hd.core.exceptions.EncryptedFileReaderWriterException: Cannot make encrypted copy for file '/Users/jim/Library/Application Support/MultiBitHD/mbhd-3a331738-d6f4a2ac-0ee81449-ba205c79-c1af7472/mbhd.wallet'
! at org.multibit.hd.core.crypto.EncryptedFileReaderWriter.encryptAndDeleteOriginal(EncryptedFileReaderWriter.java:296) ~[multibit-hd.jar:na]
! at org.multibit.hd.core.crypto.EncryptedFileReaderWriter.makeAESEncryptedCopyAndDeleteOriginal(EncryptedFileReaderWriter.java:157) ~[multibit-hd.jar:na]
! at org.multibit.hd.core.crypto.EncryptedFileReaderWriter.makeAESEncryptedCopyAndDeleteOriginal(EncryptedFileReaderWriter.java:138) ~[multibit-hd.jar:na]
! at org.multibit.hd.core.managers.WalletManager.saveWallet(WalletManager.java:1821) ~[multibit-hd.jar:na]
SNIP
! at org.multibit.hd.core.services.CoreServices.shutdownNow(CoreServices.java:205) ~[multibit-hd.jar:na]
! at org.multibit.hd.ui.controllers.MainController.shutdownCurrentWallet(MainController.java:640) ~[multibit-hd.jar:na]
! at org.multibit.hd.ui.controllers.MainController.onShutdownEvent(MainController.java:168) ~[multibit-hd.jar:na]
SNIP
Caused by: ! java.io.FileNotFoundException: /Users/jim/Library/Application Support/MultiBitHD/mbhd-3a331738-d6f4a2ac-0ee81449-ba205c79-c1af7472/mbhd.wallet (No such file or directory)
SNIP
org.multibit.hd.core.crypto.EncryptedFileReaderWriter.encryptAndDeleteOriginal(EncryptedFileReaderWriter.java:277) ~[multibit-hd.jar:na]
```

Had to force quit

**Comment by jim618** [comment]
I think this was caused by the shutdown code invoking saveWallet whilst the WalletAutoSaveListener was also doing the same.
I have improved EncryptedFileReaderWriter#encryptAndDeleteOriginal so that:
+ it is synchronized 
+ it checks the wallet file actually exists before attempting to read/encrypt/delete 

It's not easily reproducible this bug so we'll just have to watch the CrashReporter for any similar reports.

Awaiting code review and closing.
